lib/src/protocol/kawa_h1/editor.rs

```diff
@@ -0,0 +1,197 @@
+use std::{
+    net::{IpAddr, SocketAddr},
+    str::{from_utf8, from_utf8_unchecked},
+};
+
+use rusty_ulid::Ulid;
+
+use crate::{
+    pool::Checkout,
+    protocol::http::{parser::compare_no_case, GenericHttpStream},
+    Protocol,
+};
+
+#[derive(Debug)]
+pub struct HttpContext {
+    pub closing: bool,
+    pub id: Ulid,
+    pub keep_alive_backend: bool,
+    pub keep_alive_frontend: bool,
+    pub protocol: Protocol,
+    pub public_address: SocketAddr,
+    pub session_address: Option<SocketAddr>,
+    pub sticky_name: String,
+    pub sticky_session: Option<String>,
+    pub sticky_session_found: Option<String>,
+}
+
+impl kawa::h1::ParserCallbacks<Checkout> for HttpContext {
+    fn on_headers(&mut self, stream: &mut GenericHttpStream) {
+        match stream.kind {
+            kawa::Kind::Request => self.on_request_headers(stream),
+            kawa::Kind::Response => self.on_response_headers(stream),
+        }
+    }
+}
+
+impl HttpContext {
+    fn on_request_headers(&mut self, request: &mut GenericHttpStream) {
```

Comment:
please add a documenting comment to explain what it does to the generic request (and to self)